array_test.go

```diff
@@ -115,6 +115,69 @@ func TestArray_Constructors(t *testing.T) {
 	})
 }
 
+func TestArray_Decode(t *testing.T) {
+	t.Run("Decode with slice of interface", func(t *testing.T) {
+		var target []interface{}
+		testValue := []interface{}{"Foo", 123.0}
+		reporter := newMockReporter(t)
+		arr := NewArray(reporter, testValue)
+		arr.Decode(&target)
+		arr.chain.assertNotFailed(t)
+		assert.Equal(t, testValue, target)
```

Comment:
For readability, let's use the same order in all subtest, and separate them into a few blocks, e.g.:

```go
		reporter := newMockReporter(t)

		testValue := []interface{}{"Foo", 123.0}
		arr := NewArray(reporter, testValue)

		var target []interface{}
		arr.Decode(&target)

		arr.chain.assertNotFailed(t)
		assert.Equal(t, testValue, target)
```

the blocks are:

* declare reporter
* declare test value and array
* declare target and decode into it
* make assertions